### DISCUSSION TITLE
compare_models changes results at every execution

### DISCUSSION THREAD

**Discussion by leodemachado** [pycaret/pycaret#1572] [Q&A] [answered]
Hello, friends.

Does anyone know why results from compare_models() change at every execution?
I've noticed that, and I feel it as an undesirable behavior.

**Comment by Yard1** [comment, marked as answer]
Are you rerunning `setup` too?

**Reply by leodemachado** [reply]
@Yard1, yes, I am.
Sounds like there is some random piece in set_up that I am not getting.

**Reply by Yard1** [reply]
You need to set the `session_id` parameter in `setup` to some integer for reproducible results.

**Reply by leodemachado** [reply]
All right. Checked and corrected. Everything works fine now.

**Reply by leodemachado** [reply]
Guys, but after all, what is in fact this randomness component?
What has that to do with?

Thanks.

**Reply by Yard1** [reply]
`session_id` sets the random seed, which is used in many places - data shuffling, models, etc.
